docs/src/maintain/manage-releases.md

```diff
@@ -97,6 +97,43 @@ In general, we try not to do emergency releases. Even if there is a regression,
 
 The only real exception is if ESLint is completely unusable by most of the current users. For instance, we once pushed a release that errored for everyone because it was missing some core files. In that case, an emergency release is appropriate.
 
+## Major Releases
+
+This section describes additional tasks that should be done in eslint repositories and/or release infrastructure during a major release cycle. A major release cycle includes prereleases and the final release of a new major version of ESLint.
+
+For simplicity, let v9.x be the current version, and v10.0.0 the new major version.
+
+The following branches in the `eslint/eslint` repository are of importance for this release cycle:
+
+- `main`: All the latest development always happens on this branch, and that also applies to major release cycles. This means that breaking changes should be merged into this branch, along with new features, bug fixes and everything else we usually merge into this branch. Prerelease versions and the final major version will be published from this branch.
+- `next`: Branch for the latest prerelease documentation (`https://eslint.org/docs/next/`). This branch is automatically updated during the release process.
+- `v9.x-dev`: Branch for v9.x maintenance. This branch starts from the last v9.x release commit before the first v10 prerelease. All new v9.x versions (if needed) will be published from this branch.
+- `v9.x`: Branch for the ESLint v9.x documentation (`https://eslint.org/docs/v9.x/`). This branch is automatically updated during the release process. Note that this branch is needed only after the final v10.0.0 release. Until then, v9.x is still the latest version, and the documentation branch for it is `latest` (`https://eslint.org/docs/latest/`).
+
+### First Prerelease
+
+Work on the first prerelease (`alpha.0`) starts when the last planned v9.x release is concluded:
+
+- On Jenkins, check if the installed version of Node.js satisfies requirements of the new major version of ESLint. If not, install a new version of Node.js.
+- In the `eslint/eslint.org` repository, submit a PR to set `upcomingVersionPrereleaseType = "alpha"` in `tools/release-data.js`. This is to announce the prerelease in the version list on the `https://eslint.org/` homepage. The PR can be merged as soon as it is approved.
+- In the `eslint/eslint.org` repository, submit a PR to enable `/docs/next/*` proxying in `static/redirects.njk`. The PR should be reviewed and approved in time, but merged right after ESLint 10.0.0-alpha.0 is released.
+- In the `eslint/eslint` repository, submit a PR to add `legacy-peer-deps = true` in the `.npmrc` file. This is to ensure that `npm install` works in CI and locally for developers after the prerelease, because some dependencies (e.g., `@eslint-community/eslint-utils`) have `eslint` declared as a peer dependency with a range that prerelease versions like `10.0.0-alpha.0` don't satisfy. The PR can be merged as soon as it is approved.
```

Comment:
Previously, we were doing this by adding `--legacy-peer-deps` CLI flag to `npm install` commands in CI workflow files. I think this is a better solution as there's no need to update CLI commands (which we now have plenty of in the workflow files) and it also covers local installs for contributors. Let me know if this approach is not okay (e.g., if it could have some unwanted effects).